### ISSUE TITLE
Unable to generate versioned migrations: checksum mismatch

### ISSUE THREAD

**Issue by dustyhorizon** [ent/ent#2866]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Following the [docs](https://entgo.io/docs/versioned-migrations), I am unable to generate migrations using the `replay` migration mode with existing migration files. 

## Expected Behavior 🤔

With the following snippet, I expect the diff in schema to be generated and placed in `./migrations` with an updated atlas.sum.

```
const migrationsPath = "./migrations"

func main() {
	databaseURL, ok := os.LookupEnv("DATABASE_URL")
	if !ok {
		logrus.Fatalln("DATABASE_URL is not set")
		return
	}

	migrationsPathAbsDir, err := filepath.Abs(migrationsPath)
	if err != nil {
		logrus.WithError(err).Fatalln("error getting absolute path of `migrationsPath`")
	}
	if _, err := os.Stat(migrationsPathAbsDir); errors.Is(err, os.ErrNotExist) {
		err := os.Mkdir(migrationsPathAbsDir, os.ModePerm)
		if err != nil {
			logrus.WithError(err).Fatalln("error creating migrations directory")
		}
	}

	driver, err := db.NewDriver(databaseURL)
	if err != nil {
		logrus.Fatalln(err)
	}
	defer driver.Close()

	dir, err := sqltool.NewGolangMigrateDir(migrationsPathAbsDir)
	if err != nil {
		logrus.Fatalln(err)
	}

	ctx := context.Background()
	if err := migrate.NewSchema(
		entsql.OpenDB(dialect.Postgres, driver),
	).Diff(
		ctx,
		schema.WithDir(dir),
		schema.WithMigrationMode(schema.ModeReplay),
		schema.WithFormatter(sqltool.GolangMigrateFormatter),
	); err != nil {
		logrus.Fatalf("failed creating schema resources: %v", err)
	}
}
```

## Steps to Reproduce 🕹

With the previous snippet and existing migration files and valid atlas.sum file.

`atlas migrate validate` does not report any issues with the atlas.sum file.

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.19.0  |
| Ent         | 0.11.2   |
| Database    | PostgreSQL   |
| Driver      | https://github.com/jackc/pgx        |

**Comment by masseelch** [comment]
Hey @dustyhorizon, thank you for reporting this.

I tried to reproduce the issue, but couldn't. If you could use [this template](https://github.com/ent/bug) to give us a repository where I can reproduce the issue?

**Comment by dustyhorizon** [comment]
@masseelch here you go, https://github.com/dustyhorizon/ent-issue-2866

I switched up the structure just a bit but should demonstrate the issue.

Schema for `Table One` have been generated, formatting of the relevant .sql have been done, and the hashes in atlas.sum have been regenerated and validated.

The issue comes when I run `go run ./migration.go` in order to generate the schema for `Table Two` where it will error with `failed creating schema resources: validating migration directory: checksum mismatch`.

**Comment by masseelch** [comment]
Awesome! I will have a look 😊

**Comment by JoelFresh-HypeLabs** [comment]
Can confirm I'm also able to reproduce this when attempting to use the sample versioned migration code with a Postgres (v. 14.2) Database in GoLang 1.18. I'm leveraging GoLang-Migrate, atlas v.0.6.0, and ent v. 0.11.2 as well. 

What is worth noting is that if I change my database URL parameters (even to something that is invalid) I am seeing the same checksum error, so I think there may be some funny business going on beneath the scenes here. When I try to validate my atlas checksum with `atlas migrate validate...` I'm not seeing any errors either.  Any help with this is appreciated @masseelch !

**Comment by masseelch** [comment]
Hey @dustyhorizon and @JoelFresh-HypeLabs,

If you use a different migration file format than the default (`--dir-format atlas`) you currently have to provide the `--dir-format` flag to the `atlas migrate` commands as well: 

`atlas migrate validate --dir-format golang-migrate`
`atlas migrate hash --force --dir-format golang-migrate`

Since Atlas default format does not utilize "down" migration those files are currently not part of the sum-file (which is a bug I will solve). The above commands resolve your issue, until I make the changes to both Atlas and Ent. Then all files will be part of the sum-file again and you can use the commands as usual. 

I am sorry for the inconveniences, we didn't have a test case for the above :-(

**Comment by dustyhorizon** [comment]
@masseelch thanks for the workaround, can confirm it is working for me now!

**Comment by masseelch** [comment]
Once the linked PR is merged, the latest master of Ent will work again without the `--dir-format`. Though I advise to use it :-)
